vite.config.ts

```diff
@@ -0,0 +1,70 @@
+import react from '@vitejs/plugin-react';
+import alias from '@rollup/plugin-alias';
+import { join } from 'node:path';
+import { Routes } from './src/utilities/enums';
+import { VitePWA } from 'vite-plugin-pwa';
+import autoprefixer from 'autoprefixer';
+import cssNanoPlugin from 'cssnano';
+import vitePrerender from 'vite-plugin-prerender';
+import { defineConfig } from 'vite';
+import postcssFlexbugsFixes from 'postcss-flexbugs-fixes';
+
+export default defineConfig({
+	plugins: [
+		react(),
+		VitePWA({
```

Comment:
add `registerType: 'autoUpdate',` to enable the automatic generation and initialization of a service worker